README.md

```diff
@@ -73,41 +74,247 @@ function MyComponent() {
 }
 ```
 
-## TOC
+# Table of Contents
 
-- APIs
+- API
   - [GraphQLClient](#GraphQLClient)
   - [ClientContext](#ClientContext)
-  - [useClient](#useClient)
   - [useQuery](#useQuery)
   - [useManualQuery](#useManualQuery)
   - [useMutation](#useMutation)
 - Guides
   - [SSR](#SSR)
   - [Authentication](#Authentication)
-  - [Refetching a query](#Refetching-a-query)
   - [Fragments](#Fragments)
+  - [Migrating from Apollo](#Migrating-from-Apollo)
 
 ## API
 
-### `GraphQLClient`
+## `GraphQLClient`
 
-### `ClientContext`
+**Usage**:
 
-### `useClient`
+```js
+import { GraphQLClient } from 'graphql-hooks';
+const client = new GraphQLClient(config);
+```
+
+**`config`**: Object with containing configuration properites
+
+- `url` (**Required**): The url to your GraphQL server
+- `ssrMode`: Boolean - set to `true` when using on the server for server-side rendering; defaults to `false`
+- `cache`: Object with the following methods:
+  - `cache.get(key)`
+  - `cache.set(key, data)`
+  - `cache.delete(key)`
+  - `cache.clear()`
+  - `cache.keys()`
+  - `getInitialState()`
+  - See [graphql-hooks-memcache](https://github.com/nearform/graphql-hooks-memcache) as a refernce implementation
+- `fetch(url, options)`: Fetch implementation - defaults to the global `fetch` API
+- `fetchOptions`: See [MDN](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch) for info on what options can be passed
+- `headers`: Object, e.g. `{ 'My-Header': 'hello' }`
+- `logErrors`: Boolean - defaults to `true`
+- `onError({ operation, result })`: Custom error handler
+  - `operation`: Object with `query`, `variables` and `operationName`
+  - `result`: [Result Object](TODO)
+
+### `client` methods
+
+- `client.setHeader(key, value)`: Updates `client.headers` adding the new header to the existing headers
+- `client.setHeaders(headers)`: Resets `client.headers`
+- `client.logErrorResult({ operation, result })`: Default error logger; useful if you'd like to use it inside your custom `onError` handler
+- `request(operation, options)`: Make a request to your GraphQL server; returning a Promise
```

Comment:
Can you think of a use case where the developer would use this explicitly, rather than via `useQuery`?

Reply:
`useQuery` is a hook, therefore can only be used in a React component. `client.request` can be used anywhere.